packages/ag-charts-enterprise/src/box-plot/boxPlotGroup.ts

```diff
@@ -77,45 +77,84 @@ export class BoxPlotGroup extends _Scene.Group {
             lineDashOffset,
         });
 
-        outline.setProperties({ x: q1Value, y: axisValue, width: q3Value - q1Value, height: bandwidth });
-
-        boxes[0].setProperties({
-            x: q1Value,
-            y: axisValue,
-            width: Math.round(medianValue - q1Value + strokeWidth / 2),
-            height: bandwidth,
-        });
-
-        boxes[1].setProperties({
-            x: Math.round(medianValue - strokeWidth / 2),
-            y: axisValue,
-            width: Math.floor(q3Value - medianValue + strokeWidth / 2),
-            height: bandwidth,
-        });
-
-        median.setProperties({
-            x: medianValue,
-            y1: axisValue + strokeWidth,
-            y2: axisValue + bandwidth - strokeWidth,
-        });
-
-        const capY1 = axisValue + (bandwidth * (1 - cap.lengthRatio)) / 2;
-        const capY2 = axisValue + (bandwidth * (1 + cap.lengthRatio)) / 2;
-
-        caps[0].setProperties({ x: minValue, y1: capY1, y2: capY2 });
-        caps[1].setProperties({ x: maxValue, y1: capY1, y2: capY2 });
-
-        whiskers[0].setProperties({
-            x1: Math.round(minValue + whiskerProperties.strokeWidth / 2),
-            x2: q1Value,
-            y: axisValue + bandwidth / 2,
-        });
-
-        whiskers[1].setProperties({
-            x1: q3Value,
-            x2: Math.round(maxValue - whiskerProperties.strokeWidth / 2),
-            y: axisValue + bandwidth / 2,
-        });
+        const capStart = axisValue + (bandwidth * (1 - cap.lengthRatio)) / 2;
+        const capEnd = axisValue + (bandwidth * (1 + cap.lengthRatio)) / 2;
+
+        if (invertAxes) {
```

Comment:
Is there any way to avoid repeating both paths here?

E.g. could something like this reduce the repetition?

```
const categoryProp = invertAxes ? 'x' : 'y';
const valueProp = invertAxes ? 'y' : 'x';
const categoryWidthProp = invertAxes ? 'width' : 'height';
const valueDepthProp = invertAxes ? 'height' : 'width';

....

boxes[0].setProperties({
  [categoryProp]: axisValue,
  [valueProp]: q3Value,
  [categoryWidthProp]: bandwidth,
  [valueDepthProp]: Math.round(medianValue - q3Value + strokeWidth / 2),
});

```

Reply:
It's actually a problem because it isn't as simple as switching the axes, some of the math is also inverted because X starts at the bottom but Y at the top